### PR TITLE
Fixes #12777 - Use foreman's url validator

### PR DESCRIPTION
**Pull request by daviddavis**
Depends on https://github.com/theforeman/foreman/pull/2970

### CODE REVIEW

File: app/models/katello/provider.rb [hunk 1]
```diff
@@ -22,11 +22,11 @@ class Provider < Katello::Model
     validates :name, :uniqueness => {:scope => :organization_id}
     validates :provider_type, :inclusion => {:in => TYPES,
                                              :allow_blank => false, :message => "Please select provider type from one of the following: #{TYPES.join(', ')}."}
+    validates :repository_url, :url_schema => ['http', 'https', 'ftp', 'file'],
```
Comment (daviddavis) [inline app/models/katello/provider.rb]: Seems odd that we allow file protocols.

**Comment by daviddavis** [comment]
[test]

**Comment by daviddavis** [comment]
See https://github.com/theforeman/foreman/pull/2970 for more information. Would appreciate any feedback.

File: app/models/katello/repository.rb [hunk 2]
```diff
@@ -72,15 +72,14 @@ class Repository < Katello::Model
       :with => /^([a-z0-9\-_]{4,30}\/)?[a-z0-9\-_\.]{3,30}$/,
       :message => (_("must be a valid docker name"))
     }
+    validates :url, :allow_nil => :custom?, :if => :in_default_view?,
+      :url_schema => ['http', 'https', 'ftp', 'file']
```
Comment (jlsherrill) [inline app/models/katello/repository.rb]: Haven't check yet, but we need to make sure that you can use URL's with basic auth after this change. i.e.:

http://user:password@foo.example.com/bar

Comment (daviddavis) [inline app/models/katello/repository.rb]: It does. Will add a test in foreman for this.

**Comment by ehelms** [comment]
Looks like this needs a rebase and is still waiting on the corresponding PR

**Comment by ehelms** [comment]
ping @daviddavis

**Pull request closed without merging**